pkgs/os-specific/linux/drbd/driver.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, lib, fetchurl, kernel, flex, coccinelle, python3 }:
+
+stdenv.mkDerivation rec {
+  name = "drbd-${version}-${kernel.version}";
+  version = "9.2.7";
+  sha256 = "1355ns10z0fjgqsdpf09qfy01j8lg2n7zy4kclmar3s798n3mh56";
+
+  src = fetchurl {
+    url = "https://pkg.linbit.com//downloads/drbd/9/drbd-${version}.tar.gz";
+    inherit sha256;
+  };
+
+  hardeningDisable = [ "pic" ];
+
+  nativeBuildInputs = [
+    kernel.moduleBuildDependencies
+    flex
+    coccinelle
+    python3
+  ];
+
+  makeFlags = [
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "SPAAS=false"
+  ];
+
+  # 6.4 and newer provide a in-tree version of the handshake module https://www.kernel.org/doc/html/v6.4/networking/tls-handshake.html
+  installPhase = ''
+    runHook preInstall
+    install -D drbd/drbd.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/block/drbd
+    install -D drbd/drbd_transport_tcp.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/block/drbd
+    install -D drbd/drbd_transport_lb-tcp.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/block/drbd
+    install -D drbd/drbd_transport_rdma.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/block/drbd
+    ${lib.optionalString (lib.versionOlder kernel.version "6.4") ''
+      install -D drbd/drbd-kernel-compat/handshake/handshake.ko -t $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/block/drbd
+    ''}
+    runHook postInstall
+  '';
+
+  postPatch = ''
+    patchShebangs .
+    substituteInPlace Makefile --replace 'SHELL=/bin/bash' 'SHELL=${builtins.getEnv "SHELL"}'
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/LINBIT/drbd";
+    description = "LINBIT DRBD kernel module";
+    license = licenses.gpl2Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ birkb ];
+    longDescription = ''
+       DRBD is a software-based, shared-nothing, replicated storage solution
+       mirroring the content of block devices (hard disks, partitions, logical volumes, and so on) between hosts.
+    '';
+    broken = kernel.version == "6.5.13-bcachefs-unstable-2023-06-28";
```

Comment:
This is oddly specific. Where does this kernel version come from, and why would it be the only one that is broken? Maybe we can make this check more general.

Reply:
> This is oddly specific. Where does this kernel version come from, and why would it be the only one that is broken? Maybe we can make this check more general.

The [testing_bcachefs](https://github.com/NixOS/nixpkgs/pull/268394) kernel seems to be broken quite often and should be removed anyhow. I wanted to configure `meta.broken` with the generic `testing_bcachefs` name, but i have found no way to do that. Any ideas?

Reply:
`broken = !kernel.meta.broken;` is doing the job in a very generic way...

```
1 package marked as broken and skipped:
linuxKernel.packages.linux_testing_bcachefs.drbd

1 package blacklisted:
nixos-install-tools

23 packages built:
drbd linuxKernel.packages.linux_4_19.drbd linuxKernel.packages.linux_4_19_hardened.drbd linuxKernel.packages.linux_5_10.drbd linuxKernel.packages.linux_5_10_hardened.drbd linuxKernel.packages.linux_5_15.drbd linuxKernel.packages.linux_5_15_hardened.drbd linuxKernel.packages.linux_5_4.drbd linuxKernel.packages.linux_5_4_hardened.drbd linuxKernel.packages.linux_6_1.drbd linuxKernel.packages.linux_hardened.drbd linuxKernel.packages.linux_6_5.drbd linuxKernel.packages.linux_6_5_hardened.drbd linuxKernel.packages.linux_6_6.drbd linuxKernel.packages.linux_6_6_hardened.drbd linuxKernel.packages.linux_latest_libre.drbd linuxKernel.packages.linux_libre.drbd linuxKernel.packages.linux_lqx.drbd linuxKernel.packages.linux_xanmod.drbd linuxKernel.packages.linux_xanmod_latest.drbd linuxKernel.packages.linux_zen.drbd ocf-resource-agents pacemaker
```

Reply:
Is that line needed at all since https://github.com/NixOS/nixpkgs/commit/39805096cb9226f357364bdf9cc5492a8429b2ce?

I don't see the relationship between drbd and bcachefs. Even with `broken = !kernel.meta.broken;` I still do not think the drbd package should care at all if the user chooses to build broken kernels. It is not done by any other driver package either.

Reply:
Yes, since https://github.com/NixOS/nixpkgs/pull/268394 has been accepted the problem has gone away. The intention for `broken = !kernel.meta.broken;` is to not let the local `nixpkgs-review` run fail because drbd cannot be build because of a broken kernel. This should also help the central [hydra](https://github.com/NixOS/hydra) builds, or?

Reply:
If the problem has gone away, can we now drop this line?

Reply:
Yes, done.